src/hb_cache.erl

```diff
@@ -407,67 +406,79 @@ do_read_commitment(Path, Opts) ->
 
 %% @doc Load all of the commitments for a message into memory.
 read_all_commitments(Msg, Opts) ->
-    Store = hb_opts:get(store, no_viable_store, Opts),
-    UncommittedID = hb_message:id(Msg, none, Opts#{ linkify_mode => discard }),
+    Store = hb_store:scope(hb_opts:get(store, no_viable_store, Opts), local),
+    CurrentCommitments = hb_maps:get(<<"commitments">>, Msg, #{}, Opts),
+    FoundCommitments = read_all_commitments_by_store(Msg, Store, Opts),
+    NewCommitments =
+        hb_maps:merge(
+            CurrentCommitments,
+            maps:from_list(FoundCommitments)
+        ),
+    Msg#{ <<"commitments">> => NewCommitments }.
+
+read_all_commitments_by_store(Msg, Store, Opts) when not is_list(Store) ->
```

Comment:
With `read_all_commitments_by_store` we make sure all actions only occour in one `Store`, and not retried in different stores if the item isn't found in the first one.